service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CopyAzureStorageContainerDefinitionStepTest.java

```diff
@@ -130,9 +133,39 @@ void doStep_clonesDefinition() throws InterruptedException {
     assertEquals(cloned.getStorageContainerName(), destContainerName);
     assertEquals(cloned.getName(), destResourceName);
     assertEquals(cloned.getCloningInstructions(), CloningInstructions.COPY_DEFINITION);
+    assertNull(
+        cloned.getStorageAccountId(),
+        "Storage account should be null since this is a landing zone backed workspace");
     assertEquals(sharedAccountRegion, cloned.getRegion());
   }
 
+  @Test
+  void failsIfNoLandingZoneAccount() {
+    inputParams.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID, UUID.randomUUID());
+    inputParams.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_CONTAINER_NAME, "fake");
+    inputParams.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), testUser);
+
+    var sourceContainer =
+        buildContainerResource(
+            "fake-source-container",
+            "fake-source-resource-name",
+            UUID.randomUUID(),
+            UUID.randomUUID());
+
+    var step =
+        new CopyAzureStorageContainerDefinitionStep(
+            mockSamService(),
+            testUser,
+            sourceContainer,
+            controlledResourceService,
+            CloningInstructions.COPY_DEFINITION);
+
+    var ex = assertThrows(MissingRequiredFieldsException.class, () -> step.doStep(flightContext));
+    assertThat(ex.getMessage(), containsString(ControlledResourceKeys.SHARED_STORAGE_ACCOUNT));
```

Comment:
Not a big fan of the string matching here, we may want to enhance the missing fields exception to consume an optional field name but I think that's for another PR.